### ISSUE TITLE
Dynamic where{Column}() args typed as mixed variadic — column type not enforced

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#928]
## Problem

Dynamic where methods (`whereTitle('foo')`, `whereId(5)`) pass argument-type checking even when the value type contradicts the column type. The handler returns a variadic `mixed` parameter list, so `whereEmail(123)` is accepted for a `string` column.

## Where it breaks

`src/Handlers/Magic/MethodForwardingHandler.php:291-298`:

```php
// Dynamic where{Column}: provide a variadic mixed signature so Psalm's magic-method
// handler can validate argument counts. Existence itself is confirmed by Relation's
// __call; these params only govern argument arity checking.
if (self::\$enableDynamicWhere && self::isDynamicWhereMethod(\$methodName)) {
    return [new FunctionLikeParameter('args', by_ref: false, type: Type::getMixed(), is_variadic: true)];
}
```

Comment is honest about the trade-off: arity yes, type no.

## Reproduction

```php
/** @property string \$email */
class User extends Model {}

User::query()->whereEmail('valid@example.com'); // OK
User::query()->whereEmail(123);                 // Accepted — should warn (string column)
User::query()->whereId('not-an-int');           // Accepted — should warn (int column)
```

## Fix sketch

Resolve the column type via the same path `ModelPropertyHandler` uses (`getColumnType` or `\$casts` / schema), then build a parameter typed accordingly. For multi-segment forms (`whereFooAndBar`), produce one parameter per segment.

```php
foreach (\$segments as \$index => \$segment) {
    \$columnType = self::resolveColumnType(\$codebase, \$modelClass, \$segment) ?? Type::getMixed();
    \$params[] = new FunctionLikeParameter('arg' . \$index, by_ref: false, type: \$columnType);
}
```

Operator-form variants (`whereTitle('=', 'foo')`) need a fallback or `mixed` for the first arg when it's a SQL operator. Larastan handles this via a synthetic `DynamicWhereParameterReflection` that accepts operator|value polymorphism.

## Caveats

- Be tolerant of operator-form calls: `whereTitle('like', '%foo%')`. Either allow `string|VALUE` for first arg or restrict to value-only and document.
- `mixed` is acceptable when column type is unknown (no `@property`, no migration schema).

## References

- Larastan: `.alies/larastan/src/Methods/BuilderHelper.php:127-133` (returns `EloquentBuilderMethodReflection` with `DynamicWhereParameterReflection`)
- Larastan: `.alies/larastan/src/Reflection/DynamicWhereParameterReflection.php`

## Related

Multi-segment dynamic-where validation is tracked separately.
